### PR TITLE
Support direct _super call in treeFor hook

### PR DESCRIPTION
**Pull request by thoov**
Fixes: #569

### CODE REVIEW

File: packages/compat/tests/stage1.test.ts [hunk 1]
```diff
@@ -372,7 +372,7 @@ describe('stage1 build', function () {
         `
         treeFor(name) {
           if (name !== 'app') {
-            return this._super.treeFor(name);
+            return this._super(name);
```
Comment (rwjblue) [inline packages/compat/tests/stage1.test.ts:375]: Does this replace the only test for `this._super.treeFor` (shouldn't we test both things)?

Comment (thoov) [reply packages/compat/tests/stage1.test.ts:375]: No, I left the other test that @ef4 created: https://github.com/embroider-build/embroider/pull/572/files/0ba72bf537c5bd0fc6073751f7c6796d04ffdda2#diff-4397bded80f536069b8084a839df928911762318b820e9f6976baa1cc3f40538L353 which uses `this._super.treeFor`

Comment (rwjblue) [reply packages/compat/tests/stage1.test.ts:375]: Awesome, thank you!

**Comment by ef4** [comment]
Thanks!

**Pull request merged**